tests/test_parser.py

```diff
@@ -380,3 +454,46 @@ def test_tls_help(self):
         self.parser._set_defaults(None)
         help_output = self.parser._parser_tls.format_help()
         self.assert_snapshot('tls_help', help_output)
+
+
+class ParserModuleFunctionTestCase(unittest.TestCase):
+    # pylint: disable=protected-access
+    def test_create_parser(self):
+        description = 'parser description'
+        logfilename = 'logfilename'
+
+        parser = create_parser(description, logfilename)
+
+        self.assertTrue(isinstance(parser, CliParser))
+        self.assertEqual(parser._logfilename, logfilename)
+        self.assertEqual(parser._bootstrap_parser.description, description)
+
+    @mock.patch('gvmtools.parser.TLSConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_unix_socket_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test()
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.SSHConnection')
+    def test_create_tls_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test('tls', TLSConnection)
+
+    @mock.patch('gvmtools.parser.UnixSocketConnection')
+    @mock.patch('gvmtools.parser.TLSConnection')
+    def test_create_ssh_connection(
+        self, *args
+    ):  # pylint: disable=unused-argument
+        self.perform_create_connection_test('ssh', SSHConnection)
+
+        connection = create_connection('ssh', port=123)
+        self.assertTrue(isinstance(connection, SSHConnection))
+
+    def perform_create_connection_test(
+        self, connection_type='socket', connection_class=UnixSocketConnection
+    ):
+        connection = create_connection(connection_type)
+        self.assertTrue(isinstance(connection, connection_class))
```

Comment:
There is a `assertIsInstance` method in `TestCase` https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIsInstance